### PR TITLE
pkg(engine): add version in ldflags and welcome log

### PR DESCRIPTION
**Pull request by amyangfei**
### What problem does this PR solve?

Issue Number: ref #5668

### What is changed and how it works?

- log version when server-master or executor starts up
- pass necessary ldflags to engine binary
- support `-V` in `df-master` and `df-executor`

#### Tests 

 - Unit test
 - Integration test

### Release note 

```release-note
None
```

### CODE REVIEW

**Review by lance6716 — APPROVED** [review]

File: engine/cmd/executor/main.go [hunk 1]
```diff
@@ -16,15 +16,18 @@ package main
...
 	"flag"
+	"fmt"
 	"os"
 	"os/signal"
 	"strings"
 	"syscall"
 
 	"github.com/pingcap/errors"
+	"go.uber.org/zap"
+
```
Comment (lonng) [inline engine/cmd/executor/main.go:27]: Redundant blank line.

**Comment by lonng** [comment]
Rest LGTM

File: engine/cmd/master/main.go [hunk 2]
```diff
@@ -23,9 +23,10 @@ import (
 	"syscall"
 
 	"github.com/pingcap/errors"
-	"github.com/pingcap/tiflow/dm/pkg/log"
 	"go.uber.org/zap"
 
+	"github.com/pingcap/tiflow/dm/pkg/log"
```
[the remark this answers is not in the document]
Comment (lance6716) [reply engine/cmd/master/main.go:28]: agree. last week we are discussing about moving cdc's files from project's root folder to ticdc folder, after that we can use project/pkg to represent shared libraries. not a practical plan yet.

File: engine/cmd/executor/main.go (hunk 1 above)
Comment (amyangfei) [reply engine/cmd/executor/main.go:27]: The import is grouped by
- standard library
- third party library
- and import path in this repo (this is no standard about this rule, but is widely used in DM code), we need to unify this rule

File: engine/cmd/master/main.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (amyangfei) [reply engine/cmd/master/main.go:28]: Added an issue to track such work, https://github.com/pingcap/tiflow/issues/5687

**Review by lonng — APPROVED** [review]

**Pull request merged**